pkg/util/etcdutil/migrate.go

```diff
@@ -136,41 +134,36 @@ func createMigratedEtcdCluster(restclient restclient.Interface, host, podIP stri
 	return nil
 }
 
-func checkEtcdClusterUp() error {
+func checkEtcdClusterUp(restclient restclient.Interface, host string) error {
 	glog.Infof("initial delaying (30s)...")
 	time.Sleep(30 * time.Second)
 
-	// The checking does:
-	// - Trying to talk to etcd cluster via etcd service. The assumption here is that
-	//   the etcd service only selects the etcd pods newly created, excluding the boot one.
-	//   Once we can talk to it, we are sure that etcd cluster is created.
-	// - Then we list members and see if it's been reduced to 1 member. Because when we
-	//   can talk to the etcd cluster, we are certain there are 2 members at the beginning,
-	//   and will reduce to 1 eventually. That's the timeline of expected events.
-	//   As long as 1 member cluster is reached, we are certain cluster has been migrated successfully.
-	err := wait.PollImmediate(10*time.Second, 60*time.Second, func() (bool, error) {
-		cfg := clientv3.Config{
-			Endpoints:   []string{fmt.Sprintf("http://%s:2379", etcdServiceIP)},
-			DialTimeout: 5 * time.Second,
+	err := wait.Poll(10*time.Second, 60*time.Second, func() (bool, error) {
+		uri := fmt.Sprintf("%s/apis/coreos.com/v1/namespaces/kube-system/etcdclusters/%s", host, etcdClusterName)
+		res := restclient.Get().RequestURI(uri).Do()
+		if res.Error() != nil {
```

Comment:
I had to add: https://github.com/kubernetes-incubator/bootkube/compare/master...aaronlevy:sh-etcd#diff-2d3c8cde0a58be225d2f6d6b5d64059fR68 -- otherwise it would exit early if the TPR hadn't been created yet.